### PR TITLE
[Testing 2/2] Revise revisions docs

### PR DESCRIPTION
**Pull request by jieyouxu**
Update some directive docs to be more up-to-date and mention in which test suites they are supported.

### CODE REVIEW

File: src/tests/directives.md [hunk 1]
```diff
@@ -26,250 +25,242 @@ fn main() {
...
+| `check-fail`                | Building (no codegen) should fail           | `ui`, `crashes`                                  | N/A             |
+| `build-pass`                | Building should pass                        | `ui`, `crashes`, `codegen`, `incremental`[^inc1] | N/A             |
+| `build-fail`                | Building should fail                        | `ui`, `crashes`                                  | N/A             |
+| `run-pass`                  | Running the test binary should pass         | `ui`, `crashes`, `incremental`[^inc1]            | N/A             |
+| `run-fail`                  | Running the test binary should fail         | `ui`, `crashes`                                  | N/A             |
+| `ignore-pass`               | Ignore `--pass` flag                        | `ui`, `crashes`, `codegen`, `incremental`[^inc1] | N/A             |
+| `dont-check-failure-status` | Don't check exact failure status (i.e. `1`) | `ui`, `incremental`                              | N/A             |
+| `failure-status`            | Check                                       | Any `u16`                                        | N/A             |
```
Comment (jieyouxu) [inline src/tests/directives.md:78]: Oops

File: src/tests/directives.md [hunk 2]
```diff
@@ -26,250 +25,242 @@ fn main() {
...
+| `normalize-stderr-32bit`          | Normalize actual stderr (for 32-bit platforms) with a rule `"<raw>" -> "<normalized>"` before comparing against snapshot | `"<RAW>" -> "<NORMALIZED>"`, `<RAW>`/`<NORMALIZED>` is regex capture and replace syntax |                        |
+| `normalize-stderr-64bit`          | Normalize actual stderr (for 64-bit platforms) with a rule `"<raw>" -> "<normalized>"` before comparing against snapshot | `"<RAW>" -> "<NORMALIZED>"`, `<RAW>`/`<NORMALIZED>` is regex capture and replace syntax |                        |
+| `normalize-stderr-test`           | Normalize actual stderr with a rule `"<raw>" -> "<normalized>"` before comparing against snapshot                        | `"<RAW>" -> "<NORMALIZED>"`, `<RAW>`/`<NORMALIZED>` is regex capture and replace syntax |                        |
+| `normalize-stdout-test`           | Normalize actual stdout with a rule `"<raw>" -> "<normalized>"` before comparing against snapshot                        | `"<RAW>" -> "<NORMALIZED>"`, `<RAW>`/`<NORMALIZED>` is regex capture and replace syntax |                        |
```
Comment (jieyouxu) [inline src/tests/directives.md:97]: Missing test suites

File: src/tests/directives.md [hunk 3]
```diff
@@ -26,250 +25,242 @@ fn main() {
...
-* When particular debuggers are being tested: `cdb`, `gdb`, `lldb`
-* Specific [compare modes]: `compare-mode-polonius`,
-  `compare-mode-chalk`, `compare-mode-split-dwarf`,
-  `compare-mode-split-dwarf-single`
-
-The following header commands will check rustc build settings and target settings:
-
-* `needs-asm-support` — ignores if it is running on a target that doesn't have
+- A full target triple: `aarch64-apple-ios`
+- Architecture: `aarch64`, `arm`, `asmjs`, `mips`, `wasm32`, `x86_64`, `x86`,
```
Comment (Noratrieb) [inline src/tests/directives.md:130]:
```suggestion
- Architecture: `aarch64`, `arm`, `mips`, `wasm32`, `x86_64`, `x86`,
```

**Review by Noratrieb — APPROVED** [review]
MEOW

**Comment by jieyouxu** [comment]
meow

File: src/tests/directives.md [hunk 4]
```diff
@@ -26,250 +25,242 @@ fn main() {
...
+| `build_aux_docs`      | Build docs for auxiliaries as well                                                                    | All except `run-make` | N/A                                           |
+
+### Controlling outcome expectations
+
+See [Controlling pass/fail
+expectations](ui.md#controlling-passfail-expectations).
+
+| Directive                   | Explanation                                 | Supported test suites                            | Possible values |
+|-----------------------------|---------------------------------------------|--------------------------------------------------|-----------------|
+| `check-pass`                | Building (no codegen) should pass           | `ui`, `crashes`, `incremental`[^inc1]            | N/A             |
```
Comment (ehuss) [inline src/tests/directives.md:70]: Just FYI, these `[inc1]` footnotes do not have a footnote definition.

**Pull request merged**